File: DorotiTestbedApp/android/native/bridge/src/main/java/dev/doroti/bridge/DorotiNativeBridge.java
package dev.doroti.bridge;

import android.app.Activity;
import android.os.Build;
import org.json.JSONException;
import org.json.JSONObject;

public final class DorotiNativeBridge {
    private DorotiNativeBridge() { }

    public static String platformInfo() {
        try {
            return new JSONObject()
                .put("platform", "Android")
                .put("osVersion", Build.VERSION.RELEASE == null ? "unknown" : Build.VERSION.RELEASE)
                .put("bridgeVersion", "1.0.0")
                .toString();
        } catch (JSONException exception) {
            throw new IllegalStateException("Unable to create the Doroti native platformInfo payload.", exception);
        }
    }

    public static String echo(String value) {
        return DorotiNativeCore.echo(value);
    }

    public static void echoOnUiThread(
        Activity activity,
        String value,
        DorotiResultCallback callback) {
        if (activity == null) throw new IllegalArgumentException("activity");
        if (callback == null) throw new IllegalArgumentException("callback");
        activity.runOnUiThread(() -> callback.onResult(echo(value)));
    }
}

File: Doroti/templates/Doroti.Templates/content/doroti-app/android/native/bridge/src/main/kotlin/dev/doroti/bridge/DorotiNativeBridge.kt
package dev.doroti.bridge

import android.app.Activity
import android.os.Build
import org.json.JSONObject

object DorotiNativeBridge {
    @JvmStatic
    fun platformInfo(): String = JSONObject()
        .put("platform", "Android")
        .put("osVersion", Build.VERSION.RELEASE ?: "unknown")
        .put("bridgeVersion", "1.0.0")
        .toString()

    @JvmStatic
    fun echo(value: String): String = DorotiNativeCore.echo(value)

    @JvmStatic
    fun echoOnUiThread(activity: Activity, value: String, callback: DorotiResultCallback) {
        activity.runOnUiThread { callback.onResult(echo(value)) }
    }
}
